Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -4578,84 +3903,58 @@ - (void)updateCellsVisible
     }
 }
 
-- (void)updateShouldObserveDBModifications
+- (void)updateShouldObserveVMUpdates
 {
     if (!CurrentAppContext().isAppForegroundAndActive) {
-        self.shouldObserveDBModifications = NO;
+        self.shouldObserveVMUpdates = NO;
         return;
     }
 
     if (!self.isViewVisible) {
-        self.shouldObserveDBModifications = NO;
+        self.shouldObserveVMUpdates = NO;
         return;
     }
 
     if (OWSWindowManager.sharedManager.isPresentingMenuActions) {
-        self.shouldObserveDBModifications = NO;
+        self.shouldObserveVMUpdates = NO;
         return;
     }
 
-    self.shouldObserveDBModifications = YES;
+    self.shouldObserveVMUpdates = YES;
 }
 
-- (void)setShouldObserveDBModifications:(BOOL)shouldObserveDBModifications
+- (void)setShouldObserveVMUpdates:(BOOL)shouldObserveVMUpdates
```

Comment:
This is an important change.  Previously the CVC observed DB changes, but paused observation while it was hidden or the app was in the background.

Now, the CVM observes DB changes, and the CVC observes CVM updates.  When the view is hidden or the app is in the background, the CVM _continues to update_ to reflect DB changes, but the CVC ignores CVM updates.  

Ignoring changes when not visible was important for perf, but introduced some painful complications.  I think the arrangement will be considerably more convenient.